src/gt4py/eve/traits.py

```diff
@@ -172,3 +172,11 @@ def visit(self, node: concepts.RootNode, **kwargs: Any) -> Any:
             kwargs["symtable"] = kwargs["symtable"].parents
 
         return result
+
+
+class PreserveLocationWithSymbolTableTrait(VisitorWithSymbolTableTrait):
+    def visit(self, node: concepts.RootNode, **kwargs: Any) -> Any:
+        result = super().visit(node, **kwargs)
+        if hasattr(node, "location") and hasattr(result, "location"):
+            result.location = node.location
+        return result
```

Comment:
```suggestion
```
Beside the code duplication I don't think this is needed. Just inheriting from `PreserveLocation` should be fine right?